src/main/java/io/vertx/core/net/impl/SSLHelper.java

```diff
@@ -410,9 +407,9 @@ private SslHandler createHandler(SSLEngine engine, boolean client) {
           break;
         }
       }
-    } else if (verifyHost) {
+    } else if (!endpointIdentificationAlgorithm.isEmpty()) {
```

Comment:
endpointIdentificationAlgorithm could be null with a malformed option value in NetClientOptions : add a non null ckeck too